internal/api/auth/basic_auth.go

```diff
@@ -12,6 +13,44 @@ const (
 	realm                = `Basic realm="restricted"`
 )
 
+type authMode int
+
+const (
+	NoAuth authMode = iota
+	LDAPMode
+)
+
+type BasicAuth struct {
+	ldapAuth *LDAPAuth
+	mode     authMode
+}
+
+func NewBasicAuth() BasicAuth { return BasicAuth{mode: NoAuth} }
```

Comment:
Maybe we should make the user pass the `mode` parameter along with the necessary configuration explicitly here ?

Reply:
yes I totally agree.

It will be done in a dedicated PR, as I planned to do what you suggest but for the whole code.
As you known, currently the configuration is stored globally and accessed directly by all the packages.

Reply:
Actually I'll do it now as it will help solving the below comment